### DISCUSSION TITLE
Scrolling is really choppy in Safari, shows white during scroll in Chrome

### DISCUSSION THREAD

**Discussion by mmgoodnow** [jesec/flood#160] [General]
I've been noticing some pretty choppy scrolling in Safari, even in lists with 60 or so items. 

I noticed you implemented windowing in https://github.com/jesec/flood/commit/6c1b44b4d139dab0f7b3d5cf7ee0a8744d7715a2 and I think that could have something to do with it because I don't remember it being this choppy in the old jfurrow version. I understand that that's a performance optimization though.

I checked it out in Chrome and while it's not choppy, the list items that were previously hidden are white while I scroll. 

Is anybody else noticing this problem?

**Comment by jesec** [comment]
I failed to reproduce the issue. I regularly use Flood on my iPhone and Mac. There is no issue. 

Can you upload a video or screenshot to demonstrate what “choppy” means?

**Reply by mmgoodnow** [reply]
I'm using Safari for the Mac, problem doesn't happen on my iPhone. 

I'd rather not upload a video, but basically it's low FPS. Really noticeable for me when I scroll the (both condensed and expanded views) table at a slow-medium pace and watch the torrent names, then go to a [different page](https://nodejs.org/docs/latest-v13.x/api/http.html) and try scrolling there, the node docs are buttery smooth in comparison. 

Obviously it's not much of a problem; it's not inhibiting my workflow at all, just a minor annoyance.

**Reply by jesec** [reply]
There are fair amounts of rendering and reconciliation involved, so definitely it would be slower than a static page like documentation of Node.

**Comment by jesec** [comment]
Flood already has a state-of-the-art implementation of list handling. I don't think it is possible for me, at this moment, to do more about the performance of list. 

![image](https://user-images.githubusercontent.com/11585141/105796635-38729d00-5fca-11eb-9a76-ea318cd770af.png)

I did a performance profile. All rows are properly memoized. There is no wasted render when I scroll the list.

![image](https://user-images.githubusercontent.com/11585141/105796880-bc2c8980-5fca-11eb-9cb5-c4b96f68712a.png)

I think Apple does not implement smooth scrolling in macOS version of Safari. Thus, the perceived scrolling performance can be worse than anticipated. Safari on iOS feels much better.

**Comment by Wonderfall** [comment]
Yep, I can confirm the issue on Safari 14.0.3 (macOS Big Sur, 2018 15" Macbook). No issues whatsoever on my iPad Air 4.
(Unsure if it's related: Safari, at least on macOS, lacks implementation of `scroll-behavior: smooth`.)
